### DISCUSSION TITLE
New V6 Group

### DISCUSSION THREAD

**Discussion by iocoker** [fabricjs/fabric.js#8318] [Q&A] [answered]
Hi guys, thanks for this awesome framework!!

There seems to be an issue with the new group.

Here's the [Fiddle](http://jsfiddle.net/x2468nf3/25/)

1. I have a rounded rectangle clipPath in a group, and set the layout to "clip-path"
2. To retain the corners radius, I reset the scaling on the group (for this demo, I only implemented the middle right control -> mr)
3. And on scaling, I adjust the rectangle's width and call triggerLayout() to redraw.

![image](https://user-images.githubusercontent.com/81635623/192248401-190f657c-ff7b-4f57-a5c7-925e6ea6e757.png)

It works fine until you scale to about 80% of the original size, then the rectangle crops out.
![image](https://user-images.githubusercontent.com/81635623/192248201-361a6719-83b0-449e-b5ff-22002d639f14.png)

I tried adding an image, the image even bleeds out of the group, which shouldn't be, since it's meant to be clipped by the clipPath.

Please try resize with the middle right control only.

Any suggestions?

**Comment by ShaMan123** [comment, marked as answer]
I am not sure I understand your use case... it might be a much easier way achieving things with the new Group.
Anyways looks you digged into the impl.
You are touching a bare point in fabric.
Basically if you decide this is the easiest way I would suggest overriding the default control and defining a custom action. You don't really want the group to scale, do you?
So it seems... look http://jsfiddle.net/a7xc6ptn/

**Reply by ShaMan123** [reply]
My solution takes into account that the clip path resides in the plane spread by the group
Relative transform planes are the first thing you should think about when deeply customizing Group

**Comment by iocoker** [comment]
Thanks @ShaMan123, yes I don't want the group to scale, rather to hijack the control actions and redraw the clipPath.

Here's what I was trying to achieve:


https://user-images.githubusercontent.com/81635623/192533471-44305a97-aa6e-4cf0-bd36-a0be4e84fb95.mp4



As you can see, once it gets too small, the image bleeds outside of the group boundary.
CropX, CropY + width, height would have helped but I needed those rounded edges. On the other hand, clipping the image will make the border bound to the image and not the clipPath. The new group with layout = clip-path is still the best option for us.

I will go the route of implementing custom controls. Much appreciate!

**Reply by ShaMan123** [reply]
Great
Wasn't http://jsfiddle.net/a7xc6ptn/ good enough?
I didn't experience any wrong visuals

**Reply by iocoker** [reply]
No it wasn't.
![image](https://user-images.githubusercontent.com/81635623/192545990-efc22850-661c-49c9-97c0-3b014bddf1ed.png)
See, the round corners shrink / expand.

**Reply by ShaMan123** [reply]
I get you now
